src/Microsoft.Health.Dicom.Functions/Update/UpdateDurableFunction.Orchestration.cs

```diff
@@ -41,40 +43,43 @@ public partial class UpdateDurableFunction
         EnsureArg.IsNotNull(context, nameof(context)).ThrowIfInvalidOperationId();
         logger = context.CreateReplaySafeLogger(EnsureArg.IsNotNull(logger, nameof(logger)));
         ReplaySafeCounter<int> replaySafeCounter = context.CreateReplaySafeCounter(_updateMeter.UpdatedInstances);
-
+        IReadOnlyList<InstanceMetadata> instanceMetadatas;
         UpdateCheckpoint input = context.GetInput<UpdateCheckpoint>();
+        input.Partition ??= new Partition(input.PartitionKey, Partition.UnknownName);
 
         if (input.NumberOfStudyCompleted < input.TotalNumberOfStudies)
         {
             string studyInstanceUid = input.StudyInstanceUids[input.NumberOfStudyCompleted];
 
             logger.LogInformation("Beginning to update all instances new watermark in a study.");
 
-            IReadOnlyList<InstanceFileState> instanceWatermarks = await context.CallActivityWithRetryAsync<IReadOnlyList<InstanceFileState>>(
-                nameof(UpdateInstanceWatermarkAsync),
+            IReadOnlyList<InstanceMetadata> instances = await context
```

Comment:
This is unfortunately not so kosher, because if there is a partially completed orchestration where activity 1 has completed, it's output is maintained in table storage and cached on the worker node. So then when replaying, it will load that JSON as `IReadOnlyList<InstanceMetadata>`, but the JSON is actually for `IReadOnlyList<InstanceFileState>`. If you wrote the classes such their JSON could seamlessly be interchanged, then it'd be ok.

Otherwise... you may need a new orchestration lol (or divide the logic based on the presence of new input like `Partition`)